.github/workflows/engine-and-cli.yml

```diff
@@ -49,7 +49,7 @@ jobs:
   # Only run a subset of important test cases since we just need to verify basic
   # functionality rather than repeat every test already run in the other targets.
   testdev:
-    runs-on: "${{ github.repository == 'dagger/dagger' && 'dagger-v0-12-3-32c-dind' || 'ubuntu-latest' }}"
+    runs-on: "${{ github.repository == 'dagger/dagger' && 'dagger-g2-v0-12-3-32c-dind' || 'ubuntu-latest' }}"
```

Comment:
Are the `dind` runners on-demand?

Reply:
They are exclusively on demand and single tenant